File: src/main/java/net/volcano/jdacommands/model/command/arguments/implementation/codec/DoubleCodec.java
package net.volcano.jdacommands.model.command.arguments.implementation.codec;

import net.volcano.jdacommands.model.ParameterData;
import net.volcano.jdacommands.model.command.annotations.argument.Arg;
import net.volcano.jdacommands.model.command.arguments.CommandArgument;
import net.volcano.jdacommands.model.command.arguments.implementation.argument.DoubleArgument;
import net.volcano.jdacommands.model.command.arguments.interfaces.Codec;
import org.springframework.stereotype.Component;

@Component
public class DoubleCodec extends Codec<Double> {
	
	@Override
	protected CommandArgument<Double> buildArgument(ParameterData data) {
		var builder = DoubleArgument.builder();
		
		var arg = data.parameter.getAnnotation(Arg.class);
		
		if (arg != null) {
			builder.min(arg.min() != Long.MIN_VALUE ? (double) arg.min() : null);
			builder.max(arg.max() != Long.MAX_VALUE ? (double) arg.max() : null);
		}
		
		return builder.build();
	}
}

File: src/main/java/net/volcano/jdacommands/model/command/arguments/implementation/codec/DoubleCodec.kt
package net.volcano.jdacommands.model.command.arguments.implementation.codec

import net.volcano.jdacommands.model.ParameterData
import net.volcano.jdacommands.model.command.annotations.argument.Arg
import net.volcano.jdacommands.model.command.arguments.CommandArgument
import net.volcano.jdacommands.model.command.arguments.implementation.argument.DoubleArgument
import net.volcano.jdacommands.model.command.arguments.interfaces.Codec
import org.springframework.stereotype.Component

@Component
class DoubleCodec : Codec<Double?>() {

	override fun buildArgument(data: ParameterData): CommandArgument<Double?> {
		val ann = data.parameter.annotations.firstOrNull { it is Arg } as Arg?

		val min = ann?.min?.let { if (it == Long.MIN_VALUE) null else it.toDouble() }
		val max = ann?.max?.let { if (it == Long.MAX_VALUE) null else it.toDouble() }

		return DoubleArgument(min = min, max = max)
	}
}